### ISSUE TITLE
Crash in regex completion provider

### ISSUE THREAD

**Issue by mkrueger** [mono/monodevelop#5479]
```
ERROR [2018-07-23 10:29:07Z]: An unhandled exception has occurred. Terminating MonoDevelop? False
System.NullReferenceException: Object reference not set to an instance of an object
  at Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetBindingRoot (Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node) [0x0000e] in /_/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs:1395 
  at Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetBoundNodes (Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node) [0x00041] in /_/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs:1782 
  at Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetLowerBoundNode (Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node) [0x00000] in /_/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs:457 
  at Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetBoundNodes (Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode& bindableNode, Microsoft.CodeAnalysis.CSharp.BoundNode& lowestBoundNode, Microsoft.CodeAnalysis.CSharp.BoundNode& highestBoundNode, Microsoft.CodeAnalysis.CSharp.BoundNode& boundParent) [0x00039] in /_/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs:1200 
  at Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetSymbolInfoWorker (Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode node, Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel+SymbolInfoOptions options, System.Threading.CancellationToken cancellationToken) [0x00006] in /_/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs:1078 
  at Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetSymbolInfo (Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax expression, System.Threading.CancellationToken cancellationToken) [0x000e3] in /_/src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs:551 
  at Microsoft.CodeAnalysis.CSharp.CSharpExtensions.GetSymbolInfo (Microsoft.CodeAnalysis.SemanticModel semanticModel, Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax expression, System.Threading.CancellationToken cancellationToken) [0x0000a] in /_/src/Compilers/CSharp/Portable/CSharpExtensions.cs:449 
  at MonoDevelop.CSharp.Completion.Provider.RegexCompletionProvider+<GetGroups>d__5.MoveNext () [0x00245] in /home/mkrueger/work/vanilla/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/RegexCompletionProvider.cs:143 
  at MonoDevelop.CSharp.Completion.Provider.RegexCompletionProvider.ProvideCompletionsAsync (Microsoft.CodeAnalysis.Completion.CompletionContext context) [0x00465] in /home/mkrueger/work/vanilla/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/RegexCompletionProvider.cs:110 
  at Microsoft.CodeAnalysis.Completion.CompletionServiceWithProviders.GetContextAsync (Microsoft.CodeAnalysis.Completion.CompletionProvider provider, Microsoft.CodeAnalysis.Document document, System.Int32 position, Microsoft.CodeAnalysis.Completion.CompletionTrigger triggerInfo, Microsoft.CodeAnalysis.Options.OptionSet options, System.Nullable`1[T] defaultSpan, System.Threading.CancellationToken cancellationToken) [0x0013d] in /_/src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs:457 
  at Microsoft.CodeAnalysis.Completion.CompletionServiceWithProviders.ComputeNonEmptyCompletionContextsAsync (Microsoft.CodeAnalysis.Document document, System.Int32 caretPosition, Microsoft.CodeAnalysis.Completion.CompletionTrigger trigger, Microsoft.CodeAnalysis.Options.OptionSet options, Microsoft.CodeAnalysis.Text.TextSpan defaultItemSpan, System.Collections.Immutable.ImmutableArray`1[T] providers, System.Threading.CancellationToken cancellationToken) [0x00089] in /_/src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs:319 
  at Microsoft.CodeAnalysis.Completion.CompletionServiceWithProviders.GetCompletionsAsync (Microsoft.CodeAnalysis.Document document, System.Int32 caretPosition, Microsoft.CodeAnalysis.Completion.CompletionTrigger trigger, System.Collections.Immutable.ImmutableHashSet`1[T] roles, Microsoft.CodeAnalysis.Options.OptionSet options, System.Threading.CancellationToken cancellationToken) [0x002d3] in /_/src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs:249 
  at MonoDevelop.CSharp.Completion.CSharpCompletionTextEditorExtension.InternalHandleCodeCompletion (MonoDevelop.Ide.CodeCompletion.CodeCompletionContext completionContext, MonoDevelop.Ide.CodeCompletion.CompletionTriggerInfo triggerInfo, System.Int32 triggerWordLength, System.Threading.CancellationToken token, System.Boolean forceSymbolCompletion) [0x0027b] in /home/mkrueger/work/vanilla/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CSharpCompletionTextEditorExtension.cs:422
```

> VS bug [#652784](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/652784)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.745 (master)

Pull Request #5495 merged by: Mike Krüger
Author: mono
Commit: 6340fd50e024fcce144e1ddb1119b2f52504e66f (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5495 merged by: Mike Kr??ger
Author: mono
Commit: 6340fd50e024fcce144e1ddb1119b2f52504e66f (xamarin/monodevelop)
